static/js/containers/SettingsPage.js

```diff
@@ -0,0 +1,131 @@
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+import Loader from 'react-loader';
+import Grid, { Cell } from 'react-mdl/lib/Grid';
+
+import Jumbotron from '../components/Jumbotron';
+import {
+  startProfileEdit,
+  FETCH_PROCESSING,
+  clearProfile,
+} from '../actions/index';
+import ProfileFormContainer from './ProfileFormContainer';
+import PrivacyForm from '../components/PrivacyForm';
+import ProfileProgressControls from '../components/ProfileProgressControls';
+import {
+    combineValidators,
+    privacyValidation,
+} from '../util/validation';
+
+class SettingsPage extends ProfileFormContainer {
+
+  componentDidMount() {
+    this.fetchProfile();
+  }
+
+  componentDidUpdate() {
+    this.fetchProfile();
+  }
+
+  componentWillMount() {
+    this.startSettingsEdit();
+  }
+
+  componentWillUnmount() {
+    this.clearProfileEdit();
+  }
+
+  startSettingsEdit() {
+    const { dispatch } = this.props;
+    dispatch(startProfileEdit(SETTINGS.username));
+  }
+
+  render() {
+    const { profiles, ui, dispatch } = this.props;
+    let profile;
+    let childrenWithProps;
+    let privacyControlsWithProps;
+    let loaded = false;
+    let username = SETTINGS.username;
+    let preferredName = SETTINGS.name;
+    
+    if (profiles[username] !== undefined) {
+      profile = profiles[username]['profile'];
+      loaded = profiles[username].getStatus !== FETCH_PROCESSING;
+      preferredName = profile.preferredName || SETTINGS.name;
+      childrenWithProps = this.privacyWithProps(profiles[username]);
+      privacyControlsWithProps = this.privacyControlsWithProps(profiles[username]);
+    }
+
+    return (
+      <Loader loaded={loaded}>
+        <Jumbotron profile={profile} text={preferredName}>
+          <div className="card-copy">
+            { childrenWithProps }
+          </div>
+          <Grid className="profile-tab-grid privacy-form-controls">
+            <Cell col={12}>
+              { privacyControlsWithProps }
+            </Cell>
+          </Grid>
+        </Jumbotron>
+      </Loader>
+    );
+  }
+
+  privacyWithProps = profileFromStore => {
+    let { ui } = this.props;
+    let errors, isEdit, profile;
+
+    if (profileFromStore.edit !== undefined) {
+      errors = profileFromStore.edit.errors;
+      profile = profileFromStore.edit.profile;
+      isEdit = true;
+    } else {
+      profile = profileFromStore.profile;
+      errors = {};
+      isEdit = false;
+    }
+
+    return React.Children.map([<PrivacyForm />], (child) => (
+      React.cloneElement(child, {
+        profile: profile,
+        errors: errors,
+        ui: ui,
+        updateProfile: this.updateProfile.bind(this, isEdit),
+        saveProfile: this.saveProfile.bind(this, isEdit),
+        clearProfileEdit: this.clearProfileEdit,
+        fetchProfile: this.fetchProfile
```

Comment:
you could also bind `this.updateProfile` and `this.saveProfile` in the constructor, kind of preference which one you think is cleaner.